lib/middleware/proxy.js

```diff
@@ -50,6 +50,15 @@ var parseProxyConfig = function (proxies, config) {
       port = config.port
     }
     var changeOrigin = 'changeOrigin' in proxyConfiguration ? proxyConfiguration.changeOrigin : false
+    var useEnvironmentProxy = 'useEnvironmentProxy' in proxyConfiguration ? proxyConfiguration.useEnvironmentProxy : true
+    var agent
+    if (useEnvironmentProxy) {
+      var upstreamProxy = process.env[https ? 'https_proxy' : 'http_proxy']
```

Comment:
we should make sure to also read `HTTPS_PROXY` and `HTTP_PROXY` (not sure if this happens by default here)